pkg/tool/osdctl/osdctl.go

```diff
@@ -0,0 +1,191 @@
+	binarySum := fmt.Sprintf("%x", sumBytes[:])
+
+	checksumFilePath := filepath.Join(versionedDir, checksumAsset.GetName())
+	checksumBytes, err := os.ReadFile(checksumFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read osdctl checksum file '%s': %w", checksumFilePath, err)
+	}
+	checksum := strings.Split(string(checksumBytes), " ")[0]
+	if strings.TrimSpace(binarySum) != strings.TrimSpace(checksum) {
+		fmt.Printf("WARNING: Checksum for osdctl does not match the calculated value. Please retry installation. If issue persists, this tool can be downloaded manually at %s\n", osdctlBinaryAsset.GetBrowserDownloadURL())
+		// We shouldn't link this binary to latest if the checksum isn't valid
+		return nil
+	}
+
+	if err := unArchive(t.toolDir(""), filepath.Join(versionedDir, checksumAsset.GetName())); err != nil {
+		return fmt.Errorf("failed to unarchive the checksum file '%s': %w", checksumFilePath, err)
+	}
+	//Untar osdctl file and remove (so, it removes .tar.gz basically)
+
+	// Link as latest
+	latestFilePath := t.symlinkPath(latestDir)
+	err = os.Remove(latestFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to remove existing 'osdctl' binary at '%s': %w", latestDir, err)
+	}
+
+	err = os.Symlink(osdctlBinaryFilepath, latestFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to link new 'osdctl' binary to '%s': %w", latestDir, err)
+	}
+	return nil
+}
+
+// toolDir returns this tool's specific directory given the root directory all tools are installed in
+func (t *Tool) toolDir(rootDir string) string {
+	return filepath.Join(rootDir, "osdctl")
+}
+
+func (t *Tool) symlinkPath(latestDir string) string {
+	return filepath.Join(latestDir, "osdctl")
+}
+
+func (t *Tool) Remove(rootDir, latestDir string) error {
+	// Remove all binaries owned by this tool
+	toolDir := t.toolDir(rootDir)
+	err := os.RemoveAll(toolDir)
+	if err != nil {
+		return fmt.Errorf("failed to remove %s: %w", toolDir, err)
+	}
+
+	// Remove all symlinks owned by this tool
+	latestFilePath := t.symlinkPath(latestDir)
+	err = os.Remove(latestFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to remove symlinked file %s: %w", latestFilePath, err)
+	}
+	return nil
+}
+
+func unArchive(source string, destination string) error {
+	src, err := os.Open(source)
+	if err != nil {
+		return fmt.Errorf("failed to open tarball '%s': %v", source, err)
+	}
+	defer src.Close()
+	uncompressed, err := gzip.NewReader(src)
+	if err != nil {
+		return fmt.Errorf("creating gzip.Reader '%s': %v", source, err)
+	}
+	arc := tar.NewReader(uncompressed)
+	var f *tar.Header
+	for {
+		f, err = arc.Next()
+		if err != nil {
+			if err != io.EOF {
+				return err
+			}
+			break
+		}
+		if f.FileInfo().IsDir() {
+			err = os.MkdirAll(path.Join(destination, f.Name), 0755)
+			if err != nil {
+				return err
+			}
+		} else {
+			dst, err := os.OpenFile(path.Join(destination, f.Name), os.O_WRONLY|os.O_CREATE, 0755)
```

Comment:
Is there a reason for using `os.OpenFile` over `os.Create` here? In addition to being more readable and consistent with [existing methods](https://github.com/tnierman/backplane-tools/blob/main/pkg/source/github/github.go#L109-L120), the [docs seem to indicate](https://pkg.go.dev/os#OpenFile) that most of the time we'd want to avoid using `os.OpenFile`